Start adding docs via typedoc

jnicklas opened the pull request:
This is a first stab at adding docs via TypeDoc.

Unfortunately this uses a frozen in time, outdated fork of TypeDoc, found here https://github.com/TypeStrong/typedoc/pull/1184, because the output generated just isn't very nice, but is pretty great with this fork 🤷‍♂️

It seems like some other projects are also doing this. Not sure what this says about TypeDoc.

It also takes some finagling to get docs for interactors to display in a comprehensible way.

<img width="1402" alt="Screenshot 2020-10-27 at 14 41 38" src="https://user-images.githubusercontent.com/134/97309640-97593680-1862-11eb-814a-b7fa385fe7de.png">
<img width="1402" alt="Screenshot 2020-10-27 at 14 41 46" src="https://user-images.githubusercontent.com/134/97309649-9922fa00-1862-11eb-8909-c24d467de88a.png">

jnicklas committed:
Start adding docs via typedoc

jorgelainfiesta commented:
How can I preview the docs in a browser, like what command should I run? I tried browsing TypeDoc's website but I couldn't figure it out.

Gerrit0 commented:
Recommendation: switch to typedoc@beta, which is version 0.20.0-beta.#, and is under active development. I need to update the linked PR to point people towards that...

taras commented:
Thank you @Gerrit0!

jnicklas committed:
Upgrade to TypeDoc beta

jnicklas commented:
Thanks @Gerrit0, I upgraded to the beta and it works perfectly.

@jorgelainfiesta I don't have any good way of putting these docs online right now. You can run `yarn typedoc` in `packages/bigtest`, and it will output the generated docs into `packages/bigtest/docs`.

jnicklas commented:
We'll probably want to add a deploy hook for these!

jnicklas committed:
Add more docs

jnicklas committed:
Skip README

jnicklas committed:
Add dedicated start page for API docs

Doing this allows us to link stuff from the API docs and have a better
introduction. We'll have getting started guides separate anyway.

jnicklas committed:
Start adding docs for other interactor

jnicklas committed:
Add typedoc support for sub-packages

This makes it easier to work on documentation for each package
individually.

jnicklas committed:
Improve suite docs

jnicklas committed:
Document interactor class

jnicklas committed:
Add documentation for built in interactors

jnicklas committed:
Hide deprecated App interactor in docs

jnicklas committed:
Document helper functions

jnicklas committed:
Document `Page`

jnicklas committed:
Some more docs

jnicklas commented:
This is by no means done, but I think it's good enough now where we can actually merge this and start working on integrating these docs into the website.

Docs can be generated for the `@bigtest/suite`, `@bigtest/interactor` and `bigtest` packages, by running `yarn typedoc` in their respective directories.

jnicklas commented on packages/bigtest/API.md:
```diff
@@ -0,0 +1,3 @@
+# BigTest
+
+This is cool stuff {@link createInteractor}, yeah!
```
This file is just a placeholder, we'll want to expand this to link to the most relevant parts of the documentation.
